src/reuse/modules/mobile/device.ts

```diff
@@ -0,0 +1,116 @@
+    } catch (error) {
+      return this.ErrorHandler.logException(error, 'Error: During isAppInstalled', true);
+    }
+  }
+
+/**
+   * @function installAppForPlatform
+   * @memberOf mobile.device
+   * @description Install the appropriate app based on the platform the test is being executed on.
+   * @param {string} appPath - Path of the app(.apk, .ipa)
+   * @example 
+   * await mobile.device.installAppForPlatform("/path/to/your/app.apk");
+   * await mobile.device.installAppForPlatform("/path/to/your/app.ipa");
+   */
+async installAppForPlatform(appPath: string): Promise<void> {
+  const vl = this.vlf.initLog(this.installAppForPlatform);
+  const platform = await browser.capabilities.platformName;
+  try {
+      if (platform.toLowerCase() === 'android') {
+          vl.log("Installing Android app...");
+          await browser.installApp(appPath);
+          vl.log("Android app installed successfully.");
+      } else if (platform.toLowerCase() === 'ios') {
+          vl.log("Installing iOS app...");
+          await browser.installApp(appPath);
+          vl.log("iOS app installed successfully.");
+      } else {
+          throw new Error(`Unsupported platform while installing the app: ${platform}`);
+      }
+  } catch (error) {
+      this.ErrorHandler.logException(error, 'Error: During installAppForPlatform', true);
+  }
+}
+
+/**
+   * @function pressKeyByNameInAndroid
+   * @memberOf mobile.device
+   * @description Simulate pressing a hardware key on the device (e.g., back button, home button, etc.), 
+   * and it specific to Android devices
+   * @param {string} keyName - The name of the key (e.g., "back", "home", "volumeUp", etc.)
+   * @example 
+   * await mobile.device.pressKeyByNameInAndroid("back");
+   * await mobile.device.pressKeyByNameInAndroid("home");
+   * await mobile.device.pressKeyByNameInAndroid("volumeUp");
+   * await mobile.device.pressKeyByNameInAndroid("volumeDown");
+   * await mobile.device.pressKeyByNameInAndroid("power");
+   * await mobile.device.pressKeyByNameInAndroid("enter");
+   * await mobile.device.pressKeyByNameInAndroid("space");
+   * await mobile.device.pressKeyByNameInAndroid("delete");
+   * await mobile.device.pressKeyByNameInAndroid("menu");
+   * await mobile.device.pressKeyByNameInAndroid("search");
+   * await mobile.device.pressKeyByNameInAndroid("camera");
+   * await mobile.device.pressKeyByNameInAndroid("focus");
+   * await mobile.device.pressKeyByNameInAndroid("notification");
+   */
+async pressKeyByNameInAndroid(keyName: string): Promise<void> {
+  const vl = this.vlf.initLog(this.pressKeyByNameInAndroid);
+
+  const keyCodeMap: { [key: string]: number } = {
+      back: 4,
+      home: 3,
+      volumeUp: 24,
+      volumeDown: 25,
+      power: 26,
+      enter: 66,
+      space: 62,
+      delete: 67,
+      menu: 82,
+      search: 84,
+      camera: 27,
+      focus: 80,
+      notification: 83,
+  };
+
+  try {
+    const keyCode = keyCodeMap[keyName.toLowerCase()];
```

Comment:
add if-else for arg type (string, number) to support both, name and number key

Reply:
Added a new function that accepts a number as a parameter as a key code.
https://github.com/SAP/wdio-qmate-service/blob/2cfce035e869229454bee0785ac403e49b97ab07/src/reuse/modules/mobile/android.ts